src/data_structures/segment_tree.md

```diff
@@ -158,11 +158,12 @@ The green vertices are the vertices that we visit and update.
 
 The main consideration is how to store the Segment Tree.
 Of course we can define a $\text{Vertex}$ struct and create objects, that store the boundaries of the segment, its sum and additionally also pointers to its child vertices.
-However this requires storing a lot of redundant information.
-We will use a simple trick, to make this a lot more efficient. 
-We only store the sums in an array.
+However, this requires storing a lot of redundant information in the form of pointers.
+We will use a simple trick to make this a lot more efficient of using an _implicit data structure_: Only storing the sums in an array.
```

Comment:
Is it "... more efficient **by** using ... "?

Reply:
oh, "of using" was supposed to be for "we will use a simple trick", but "by using" reads more naturally